recursive_binary_search/binary_search.cpp

```diff
@@ -0,0 +1,58 @@
+/* Program - Binary search over an array
+   Author - Harsh Khatore
+   Language - C++
+   Date - 11.06.2017
+*/
+
+#include <iostream>
+
+using namespace std;
+
+/* function to perform binary search
+input parameters:
+ar - user-input array.
+l - left index of array.
+r - right index of array.
+ele - element to be searched.
+*/
+int binarySearch(int *ar, int l, int r, int ele)
+{
+   if (r >= l)
+   {
+        int mid = l + (r - l)/2;    //calculate mid point of the array
+ 
+        // If the element is present at the middle itself
+        // +1 so that it is easier to read
+        if (ar[mid] == ele)  return mid+1;   
```

Comment:
Add a block here, even though it has 1 statement. Do the same in other parts of code.